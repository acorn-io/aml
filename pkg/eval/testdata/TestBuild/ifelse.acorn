x: if y == 1 {
	0
} else if y == 2 {
	1
} else {
	2
}