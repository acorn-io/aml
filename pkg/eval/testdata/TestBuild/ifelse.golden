&eval.File{Body: &eval.Struct{
	Position: eval.Position{
		Filename: "ifelse.acorn",
		Line:     1,
		Column:   1,
	},
	Fields: []eval.Field{&eval.KeyValue{
		Key: eval.FieldKey{
			Key: eval.Value{
				Value: value.String("x"),
			},
			Pos: eval.Position{
				Filename: "ifelse.acorn",
				Line:     1,
				Column:   1,
			},
		},
		Value: &eval.If{
			Condition: &eval.Op{
				Operator: value.Operator("=="),
				Left: &eval.Lookup{
					Pos: eval.Position{
						Filename: "ifelse.acorn",
						Offset:   6,
						Line:     1,
						Column:   7,
					},
					Key: "y",
				},
				Right: eval.Value{Value: value.Number("1")},
				Pos: eval.Position{
					Filename: "ifelse.acorn",
					Offset:   8,
					Line:     1,
					Column:   9,
				},
			},
			Value: &eval.Struct{
				Position: eval.Position{
					Filename: "ifelse.acorn",
					Offset:   13,
					Line:     1,
					Column:   14,
				},
				Fields: []eval.Field{&eval.Embedded{
					Pos: eval.Position{
						Filename: "ifelse.acorn",
						Offset:   16,
						Line:     2,
						Column:   2,
					},
					Expression: eval.Value{Value: value.Number("0")},
				}},
			},
			Else: &eval.If{
				Condition: &eval.Op{
					Operator: value.Operator("=="),
					Left: &eval.Lookup{
						Pos: eval.Position{
							Filename: "ifelse.acorn",
							Offset:   28,
							Line:     3,
							Column:   11,
						},
						Key: "y",
					},
					Right: eval.Value{Value: value.Number("2")},
					Pos: eval.Position{
						Filename: "ifelse.acorn",
						Offset:   30,
						Line:     3,
						Column:   13,
					},
				},
				Value: &eval.Struct{
					Position: eval.Position{
						Filename: "ifelse.acorn",
						Offset:   35,
						Line:     3,
						Column:   18,
					},
					Fields: []eval.Field{&eval.Embedded{
						Pos: eval.Position{
							Filename: "ifelse.acorn",
							Offset:   38,
							Line:     4,
							Column:   2,
						},
						Expression: eval.Value{Value: value.Number("1")},
					}},
				},
				Else: &eval.Struct{
					Position: eval.Position{
						Filename: "ifelse.acorn",
						Offset:   47,
						Line:     5,
						Column:   8,
					},
					Fields: []eval.Field{&eval.Embedded{
						Pos: eval.Position{
							Filename: "ifelse.acorn",
							Offset:   50,
							Line:     6,
							Column:   2,
						},
						Expression: eval.Value{Value: value.Number("2")},
					}},
				},
			},
		},
		Pos: eval.Position{
			Filename: "ifelse.acorn",
			Line:     1,
			Column:   1,
		},
	}},
}}
